File: branches/IVEF_0_1_RELEASE/ivef-objc/Classes/IVEFLoginResponse.m

#import "IVEFLoginResponse.h"


@implementation IVEFLoginResponse

- (id) init {
    self = [super init];
    if (self != nil) {
        m_reasonPresent = false;
        m_dateFormatter = [[NSDateFormatter alloc] init];
        [m_dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSS"];
    }
    return self;
}

-(void) setMsgId:(NSString *) val {

    [m_msgId release];
    m_msgId = val;
    [m_msgId retain];
}

- (NSString *) getMsgId {

    return m_msgId;
}

-(void) setResult:(int) val {

    if ( ( val != 1 ) &&
         ( val != 2 ) )
        return;
    m_result = val;
}

- (int) getResult {

    return m_result;
}

-(void) setReason:(NSString *) val {

    m_reasonPresent = true;
    [m_reason release];
    m_reason = val;
    [m_reason retain];
}

- (NSString *) getReason {

    return m_reason;
}

-(bool) hasReason {

    return m_reasonPresent;
}

-(void) setAttributes:(NSDictionary *)attributeDict {

        for (NSString *key in attributeDict) {
            if ([key isEqualToString: @"MsgId"]) {
                NSString *val = [attributeDict objectForKey: key];
                [self setMsgId: val];
            }
            else if ([key isEqualToString:@"Result"]) {
                NSString *value = [attributeDict objectForKey: key];
                int val = [value intValue];
                [self setResult: val];
            }
            else if ([key isEqualToString:@"Reason"]) {
                NSString *val = [attributeDict objectForKey: key];
                [self setReason: val];
            }
        }
}

-(NSString *) XML {

    NSMutableString *xml = [NSMutableString stringWithString:@"<LoginResponse"];
    [xml appendString: @" MsgId=\""];
    [xml appendString: [m_msgId encode]];
    [xml appendString: @"\""];
    [xml appendString: @" Result=\""];
    [xml appendString: [NSString stringWithFormat:@"%d", m_result]];
    [xml appendString: @"\""];
    if ( [self hasReason] ) {
        [xml appendString: @" Reason=\""];
        [xml appendString: [m_reason encode]];
        [xml appendString: @"\""];
    }
    [xml appendString:@">\n"];
    [xml appendString: @"</LoginResponse>\n"];
    return xml;
}

-(NSString *) encode: (NSString *) input {

    NSMutableString *str = [[[NSMutableString alloc] initWithString: input] autorelease];

    [str replaceOccurrencesOfString: @"&" withString: "&amp;") options: nil searchRange: NSMakeRange(0, [str length])];
    [str replaceOccurrencesOfString: @"<" withString: "&lt;") options: nil searchRange: NSMakeRange(0, [str length])];
    [str replaceOccurrencesOfString: @">" withString: "&gt;") options: nil searchRange: NSMakeRange(0, [str length])];
    [str replaceOccurrencesOfString: @"\"" withString: "&quot;") options: nil searchRange: NSMakeRange(0, [str length])];

    return str;
}

-(NSString *) stringValueWithLead: (NSString *) lead {

    NSMutableString *str = [[[NSMutableString alloc] init] autorelease];
    [str setString: [lead stringByAppendingString:@"LoginResponse\n"]];
    [str appendString: [lead stringByAppendingString: @" "]];
    [str appendString: @"MsgId=\""];
    [str appendString: m_msgId];
    [str appendString: @"\"\n"];

    [str appendString: [lead stringByAppendingString: @" "]];
    [str appendString: @"Result=\""];
    [str appendString: [NSString stringWithFormat:@"%d", m_result]];
    [str appendString: @"\"\n"];

    if ( [self hasReason] ) {
        [str appendString: [lead stringByAppendingString: @" "]];
        [str appendString: @"Reason = \""];
        [str appendString: m_reason];
    }
    return str;
}


@end

File: ivef-lib/trunk/ivef-qt/IVEFLoginResponse.cpp

#include "IVEFLoginResponse.h"

LoginResponse::LoginResponse() {

    m_msgId = "";
    m_result = 0;
    m_reason = "";
    m_reasonPresent = false;
}

LoginResponse::LoginResponse(const LoginResponse &val) : QObject() {

    m_msgId = val.m_msgId;
    m_result = val.m_result;
    m_reasonPresent = val.m_reasonPresent;
    m_reason = val.m_reason;
}

LoginResponse & LoginResponse::operator=(const LoginResponse &val) {

    m_msgId = val.m_msgId;
    m_result = val.m_result;
    m_reasonPresent = val.m_reasonPresent;
    m_reason = val.m_reason;
    return *this;
}

QString LoginResponse::encode( QString str) {

    str.replace('&', "&amp;");
    str.replace('<', "&lt;");
    str.replace('>', "&gt;");
    str.replace('"', "&quot;");
    return str;
}

void LoginResponse::setMsgId(QString val) {

    m_msgId = val;
}

QString LoginResponse::getMsgId() const {

    return m_msgId;
}

void LoginResponse::setResult(int val) {

    if ( ( val != 1 ) &&
         ( val != 2 ) )
        return;
    m_result = val;
}

int LoginResponse::getResult() const {

    return m_result;
}

void LoginResponse::setReason(QString val) {

    m_reasonPresent = true;
    m_reason = val;
}

QString LoginResponse::getReason() const {

    return m_reason;
}

bool LoginResponse::hasReason() {

    return m_reasonPresent;
}

QString LoginResponse::toXML() {

    QString xml = "<LoginResponse";
    xml.append(" MsgId=\"" + encode (m_msgId) + "\"");
    xml.append(" Result=\"" + QString::number(m_result) + "\"");
    if ( hasReason() ) {
        xml.append(" Reason=\"" + encode (m_reason) + "\"");
    }
    xml.append(">\n");
    xml.append( "</LoginResponse>\n");
    return xml;
}

QString LoginResponse::toString(QString lead) {

    QString str = lead + "LoginResponse\n";
    str.append( lead + "    MsgId = " + m_msgId + "\n");
    str.append( lead + "    Result = " + QString::number(m_result) + "\n");
    if ( hasReason() ) {
        str.append( lead + "    Reason = " + m_reason + "\n");
    }
    return str;
}

